prow/config/config.go

```diff
@@ -507,6 +517,30 @@ func validatePresets(name string, labels map[string]string, spec *v1.PodSpec, pr
 	return nil
 }
 
+func validatePodSpec(name string, jobType kube.ProwJobType, spec *v1.PodSpec) error {
+	if spec == nil {
+		return nil
+	}
+
+	if len(spec.InitContainers) != 0 {
```

Comment:
Can we put a TODO to relax these restrictions later?

Reply:
That would require a radical redesign of what a test is and how `plank` interacts with them, so unless a `//TODO` to do architecture+design makes sense I'd leave it as-is?